parlai/core/metrics.py

```diff
@@ -82,14 +83,17 @@ def aggregate_task_reports(reports, tasks, micro=True):
     total_exs = sum(exs.values())
     total_report['exs'] = total_exs
     for metric, task_vals in metrics.items():
-        if micro:
-            # average over the number of examples
-            vals = [task_vals[task] * exs[task] for task in tasks]
-            total_report[metric] = round_sigfigs(sum(vals) / total_exs, 4)
-        else:  # macro
-            # average over tasks
-            vals = task_vals.values()
-            total_report[metric] = round_sigfigs(sum(vals) / len(vals), 4)
+        if all([isinstance(v, Number) for v in task_vals.values()]):
+            if micro:
+                # average over the number of examples
+                vals = [task_vals[task] * exs[task] for task in tasks]
+                total_report[metric] = round_sigfigs(sum(vals) / total_exs, 4)
+            else:  # macro
+                # average over tasks
+                vals = task_vals.values()
+                total_report[metric] = round_sigfigs(sum(vals) / len(vals), 4)
+        else:
```

Comment:
i don't think this is the right behavior here. if it's a number, let's aggregate it, otherwise let's ignore it (no "else")

Reply:
For the image chat model, some of the `task_vals` are going to be dicts corresponding to which round is being computed; e.g., the metrics dict might look like the following:

```
metrics = {
   'first_round': {
          'hits@1': ...
           ...
     },
     'second_round': {...},
     'third_round': {...},
}
```

Reply:
As far as these separated metrics for each task still get printed we can get rid of the else. Just want to make sure we don't lose information here : )

Reply:
as @dexterju said, they will still be printed, they just won't be aggregated across different tasks. is this ok @klshuster ?

Reply:
ahh yeah should be fine